miner/worker.go

```diff
@@ -249,6 +250,25 @@ func (miner *Miner) prepareWork(genParams *generateParams, witness bool) (*envir
 		// configure the gas limit of pending blocks with the miner gas limit config when using optimism
 		header.GasLimit = miner.config.GasCeil
 	}
```

Comment:
Just a thought: it would be nice to have this code in the `op-node` to reduce diff in the EL but I think its way more simple to put it here

Reply:
added a "ValidateHoloceneNonce" to eip1559.go to clean this up a little bit.

Reply:
The spec requires us to do these checks and translations of zero nonce into non-zero in the EL, so the CL isn't required to set a non-zero nonce with actual values.